tests/networkpolicy_test.go

```diff
@@ -223,17 +320,44 @@ func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
 	}, 5*time.Second, 1*time.Second).ShouldNot(Succeed())
 }
 
-func waitForNetworkPolicyDeletion(policy *v1network.NetworkPolicy) {
+func createNetworkPolicy(policy *networkv1.NetworkPolicy) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	By(fmt.Sprintf("Create %s networkpolicy", policy.Name))
+	_, err = virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Create(policy)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed creating network policy %s/%s", policy.Namespace, policy.Name))
+}
+
+func waitForNetworkPolicyDeletion(policy *networkv1.NetworkPolicy) {
 	if policy == nil {
 		return
 	}
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &v13.DeleteOptions{})).To(Succeed())
+	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &metav1.DeleteOptions{})).To(Succeed())
 	EventuallyWithOffset(1, func() error {
-		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, v13.GetOptions{})
+		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, metav1.GetOptions{})
 		return err
-	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())))
+	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())))
+}
+
+func PingHTTPEventually(vmiFrom, vmiTo *v1.VirtualMachineInstance, port int) AsyncAssertion {
+	const curlCheckCmd = "curl --head %s --connect-timeout 5\n"
+	return Eventually(func() error {
+		for _, ip := range vmiTo.Status.Interfaces[0].IPs {
+			if netutils.IsIPv6String(ip) {
+				ip = "[" + ip + "]"
+			}
+			url := fmt.Sprintf("http://%s:%d", ip, port)
+			curlCheck := fmt.Sprintf(curlCheckCmd, url)
+			err := tests.VmiConsoleRunCommand(vmiFrom, curlCheck, 10*time.Second)
+			if err != nil {
+				return errors.Wrapf(err, "failed HTTP ping from vmi(%s/%s) to url(%s) at vmi(%s/%s)", vmiFrom.Namespace, vmiFrom.Name, url, vmiTo.Namespace, vmiTo.Name)
```

Comment:
I Go 1.13 there is a standard alternative to this, see [here](https://blog.golang.org/go1.13-errors).
I would prefer that over using a 3rd party package which is not used in our test.

Reply:
Done